mesheryctl/internal/cli/root/system/start.go

```diff
@@ -341,14 +341,24 @@ func start() error {
 		// downloaded required files successfully now apply the manifest files
 		log.Info("Starting Meshery...")
 
-		// apply the adapters mentioned in the config.yaml file to the Kubernetes cluster
-		err = utils.ApplyManifestFiles(manifests, RequestedAdapters, kubeClient, false, false)
+		spinner := utils.CreateDefaultSpinner("Deploying Meshery on Kubernetes.", "...Meshery deployed on Kubernetes.")
+		spinner.Start()
 
-		if err != nil {
-			return err
+		deadline := time.Now().Add(20 * time.Second)
+
+		for !(time.Now().After(deadline)) {
+			// apply the adapters mentioned in the config.yaml file to the Kubernetes cluster
+			err := utils.ApplyManifestFiles(manifests, RequestedAdapters, kubeClient, false, false)
```

Comment:
`ApplyManifestFiles` will just apply the manifests. It does not actually check if the pods are running. For that we need to check the status of the pods in `meshery` namespace.